### ISSUE TITLE
`cargo vendor` should offer the ability to omit the `tests/` directory

### ISSUE THREAD

**Issue by joshlf** [rust-lang/cargo#13474]
### Problem

As reported in https://github.com/google/zerocopy/issues/775, some users want to avoid having to vendor a crate's `test/` directory. It would be good if there were a way to do this natively in `cargo vendor`, e.g. via a `--no-tests-dir` flag or similar.

**Comment by epage** [comment]
I can definitely see the two sides to this
- At least for crater, having `.crate` files include tests is helpful
- For vendoring, tests cause more vendoring churn
- Not all vendoring people may want this policy

I suspect there is enough convention around `tests/` and `benches/` that we could have built-in support for stripping them.  I wonder however, if it'd work better to have allow the project doing the vendoring to provide some kind of exclusion file.  Figuring out the right balance for package selections would be interesting.

@emilio (hopefully this pings you) as the actual person asking about limiting vendoring, any thoughts?

**Comment by joshlf** [comment]
> I suspect there is enough convention around `tests/` and `benches/` that we could have built-in support for stripping them.  I wonder however, if it'd work better to have allow the project doing the vendoring to provide some kind of exclusion file.  Figuring out the right balance for package selections would be interesting.

Is your thinking that we (the crate authors) would specify "these files should be put in the crate file but not kept during vendoring"? If so, that'd work for us!

**Comment by epage** [comment]
No, that we could give greater control to the person doing the vendoring.

I worry that the crate authors don't have enough information about what the person doing the vendoring is going to need, or not.

**Comment by joshlf** [comment]
Ah, I misread your comment. That obviously works for us too 🙂

**Comment by emilio** [comment]
Something like an exclusion list might be useful for other use cases apart from excluding tests, like excluding configurations that we don't build for or such... But yeah that seems like a reasonable solution for this, particularly if it makes it easy to exclude whole directories (a gitignore(-like?) syntax might be enough).

@glandium, any thoughts?

**Comment by glandium** [comment]
One may want to exclude files for specific crates only, too.
